docs/development-guide.md

```diff
@@ -1,128 +1,158 @@
+#### 4.  Find the IP address of your application, then visit the application on your browser to confirm installation.
+```sh
+kubectl get service frontend-external
+```
+---
 
-    - To launch **Minikube** (tested with Ubuntu Linux). Please, ensure that the
-       local Kubernetes cluster has at least:
-        - 4 CPUs
-        - 4.0 GiB memory
-        - 32 GB disk space
+## Option 2 - Local Cluster
 
-      ```shell
-      minikube start --cpus=4 --memory 4096 --disk-size 32g
-      ```
+#### 1. A local Kubernetes cluster can be launched with one of the following tools:
 
-    - To launch **Docker for Desktop** (tested with Mac/Windows). Go to Preferences:
-        - choose “Enable Kubernetes”,
-        - set CPUs to at least 3, and Memory to at least 6.0 GiB
-        - on the "Disk" tab, set at least 32 GB disk space
+- **Minikube** (tested with Ubuntu Linux).
+  - Please, ensure that the local Kubernetes cluster has at least:
+    - 4 CPUs
+    - 4.0 GiB memory
+    - 32 GB disk space
 
-    - To launch a **Kind** cluster:
+    ```shell
+    minikube start --cpus=4 --memory 4096 --disk-size 32g
+    ```
 
-      ```shell
-      kind create cluster
-      ```
+- **Docker for Desktop** (tested with Mac/Windows).
+  - Go to Preferences:
+    - Choose “Enable Kubernetes”,
+    - Set CPUs to at least 3, and Memory to at least 6.0 GiB
+    - On the "Disk" tab, set at least 32 GB disk space
 
-2. Run `kubectl get nodes` to verify you're connected to the respective control plane.
+- **Kind** cluster:
+  - Use the basic cluster configuration provided at `local/kind-cluster.yaml`
+    ```shell
+    kind create cluster --config ./local/kind-cluster.yaml
+    kind get clusters
+    kubectl config use-context kind-hipster
+    ```
 
-3. Run `skaffold run` (first time will be slow, it can take ~20 minutes).
-   This will build and deploy the application. If you need to rebuild the images
+#### 2. Run `kubectl get nodes` to verify you're connected to the respective control plane.
+
+#### 3. Create a _GCP IAM Service Account_ in your Google Cloud Project
+- Create a new service account
+- Make sure the service account has the following roles _([see this image](img/service-account.png))_
+  ```sh
+  Cloud Trace Agent
+  Cloud Profiler Agent
+  Cloud Debugger Agent
+  Monitoring Metric Writer
+  ```
+- Download the **service account key file** to your local computer
+
+#### 4. Update the `gcp-service-account.yaml` secret with the downloaded key file data
+- Replace the `PATH_TO_DOWNLOADED_SERVICE_ACCOUNT_KEY_FILE` placeholder with the
+to the downloaded file and execute the following commands:
+  ```sh
+  # to be run from the root diectory of this repository
+  ENCODED=$(base64 <PATH_TO_DOWNLOADED_SERVICE_ACCOUNT_KEY_FILE>)
+  sed -i -e "s/KEY_FILE_CONTENT/"$ENCODED"/" local/gcp-service-account.yaml
+  ```
+#### 5. Run `skaffold run -p local`
+> 🎯 &nbsp;&nbsp;First time will be slow; it can take ~20 minutes. <br>
```

Comment:
Maybe explain that this is because docker doesn't have any cached image layers yet for the 11 services, so it has to build all of them from scratch

Reply:
Added extra descriptions